nvflare/lighter/impl/static_file.py

```diff
@@ -482,80 +478,39 @@ def _validate_host_name_against_listener(
-                    conn_host = server.name
-                else:
-                    ct = participant.get_connect_to()
-                    if ct:
-                        if not ct.name or ct.name == server.name:
-                            # connect to server directly - no relay
-                            err = self._check_host_name_against_server(ct.host, server)
-                            if err:
-                                ctx.warning(
-                                    f"connect_to.host '{ct.host}' in {participant.subject} may be invalid: {err}"
-                                )
-                            conn_host = ct.host
-                        else:
-                            # uses relay
-                            # since relay config is done in relay__resources.json, host name doesn't matter here.
-                            # we use default server here
-                            conn_host = server.get_default_host()
-
-                        if ct.port:
-                            # override fl_port
-                            fl_port = ct.port
-                    else:
-                        # connect_to is not explicitly specified: use the server's name by default
-                        # Note: by doing this dynamically, we guarantee the sp_end_point to be correct, even if the
-                        # project.yaml does not specify the default server host correctly!
-                        conn_host = server.get_default_host()
-
-                # change the sp_end_point to use conn_host
-                agent_args = overseer_agent.get("args")
-                if agent_args:
-                    sp_end_point = agent_args.get("sp_end_point")
-                    if sp_end_point:
-                        # format of the sp_end_point:  server_host_name:fl_port:admin_port
-                        agent_args["sp_end_point"] = f"{conn_host}:{fl_port}:{admin_port}"
-
-            overseer_agent.pop("overseer_exists", None)
-            config["overseer_agent"] = overseer_agent
+                # uses relay
+                # since relay config is done in relay__resources.json, host name doesn't matter here.
+                # we use default server here
+                conn_host = server.get_default_host()
+
+            if ct.port:
+                port = ct.port
+        else:
+            # connect_to is not explicitly specified: use the server's name by default
+            # Note: by doing this dynamically, we guarantee the sp_end_point to be correct, even if the
+            # project.yaml does not specify the default server host correctly!
+            conn_host = server.get_default_host()
+        return conn_host, port
 
     def _build_admin(self, admin: Participant, ctx: ProvisionContext):
+        self.prepare_admin_config(admin, ctx)
+
         dest_dir = ctx.get_kit_dir(admin)
         admin_port = ctx.get(CtxKey.ADMIN_PORT)
```

Comment:
Should rename this to port or server_port, admin_port may cause confusion.

Reply:
Don't think so.